The Supermatter Zaps with honkers and blackjack [Ready]

LemonInTheDark opened the pull request:
## About The Pull Request

Changing some shit up with the sm to make it more deadly/interesting.
It'll cause higher damage and harm objects/emp machines at a lower power level, and zap you like a tesla/blow a hole in the floor at a higher one.
Along with this I've made grounding rods 100% effective so we don't fuck the universe.

[Blue](https://cdn.discordapp.com/attachments/326831214667235328/669431768297897984/unknown.png)
[Red](https://cdn.discordapp.com/attachments/326831214667235328/669432240270344223/unknown.png)

## Why It's Good For The Game

Makes things more rewarding and fun, adds some more spice to our death rock.

## Changelog
:cl: LemonInTheDark; Code. OFC; Idea stream and blessed sprites 
tweak: Added some pretty colors. Red and blue to be exact.
balance: The SM's bolts are more deadly now, they can do more damage, emp stuff, and blow shit up. Be very careful with high power setups. By high power I mean 2k eer above the zap threshold. This comes with a buff to tesla coils and the zaps, at least in the higher colors. Go to town.
/:cl:

81Denton commented:
This looks really neat. For the EMP zaps (correct me if I'm wrong), I'm worried that a random EMP might pulse the nearby air alarm's panic siphon wire and delam the crystal faster than an Event Hall player could. Should be fine if the engine air alarms get replaced with an EMP shielded subtype though.

AtlasRules commented:
add green zaps so I can finally have a resonance cascade in spessman

trollbreeder commented:
> add green zaps so I can finally have a resonance cascade in spessman

What if delamming the SM by putting it in a vacuum made it spew out portals that contained random mobs (goliaths slimes hivelords basilisks syndicate men etc)?

MishaTarkus commented:
+1 for green zaps, we Half Life now

LemonInTheDark commented:
> 
> 
> I'm worried that a random EMP might pulse the nearby air alarm's panic siphon wire and delam the crystal faster than an Event Hall player could. Should be fine if the engine air alarms get replaced with an EMP shielded subtype though.

In my limited testing the emp's act more as a deterrent to higher power setups, as they force stop co2 input, and neuter the engine. There is a chance that this could cause a buildup of plasma and thus a fire, but it's a slow process, and I consider the debugging that it forces players to do a feature.

I will do some more expansive testing on the worst case senarios, but I think I'm happy with it as it is.

LemonInTheDark commented:
> 
> 
> GREEN ZAPS

...... Not sure what to use them for, I quite like the blue ones for emps, any suggestions?
>
>What if delamming the SM by putting it in a vacuum made it spew out portals that contained random mobs (goliaths slimes hivelords basilisks syndicate men etc)?

Low pressure delams are really..... hard to do. It's currently a baby mode setup. If/when that gets expanded on some related delam/effect would be great.

CRITAWAKETS commented:
resonance cascade should be obtained by combining the singuloth delam and the tesla delam, and it releases simplemobs across the whole station in a portal storm as the SM arcs become extremely unstable then the SM explodes.

PKPenguin321 commented:
I never thought I'd see a resonance cascade, much less create one...

JAremko commented:
> resonance cascade should be obtained by combining the singuloth delam and the tesla delam, and it releases simplemobs across the whole station in a portal storm as the SM arcs become extremely unstable then the SM explodes.

It's easier to do both mole and power delem then only mole. You can add extra delamination type for heat only.  You can use it for the mob portals meme.

LemonInTheDark commented:
> 
> It's easier to do both mole and power delem than only mole. You can add extra delamination type for heat only and use it for the mob portals meme.

That's true, but power is somewhat a function of heat at this point anyhow.
Also if we want a green lightning situation I'd prefer to have that be definite.
I like the idea of using portals to resolve the delam conflict between teslas and singulos, I'd just need a sprite for them, because as we all know, I am a master artist.

JAremko commented:
Mole can be portal. And power + mole can be singulo.  This way normal sabotages still gonna be achievable. And something more memy like the portal thingy can require babysitting. 

Also it's not clear for how long the portals will be open.   Can crew just defeat all the mobs and keep playing? I imagine the most epic way to implement it would include moving the space station into separate dimension (maybe just change the sky sprites) + spawning a boss that you have to defeat in order to get back into your dimension - then you can call a shuttle and go home.

JAremko commented:
For now the new boss can bee some kind of a blob that spawns from SM it won't even need much coding I imagine :thinking:

81Denton commented:
Resonance cascade shouldn't be too terribly hard to implement since we already have the portal storm event

JAremko commented:
I think it will help with testing if this PR included extremely basic merge of grounding rods and tesla coils. Basically same machine but if it on top of a wire it will also produce energy. 

See [this thread](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=25244) for reference

JAremko commented:
Here's IMHO better sprites + the green one.

[beam.zip](https://github.com/tgstation/tgstation/files/4100890/beam.zip)

LemonInTheDark commented:
> 
> 
> I think it will help with testing if this PR included extremely basic merge of grounding rods and tesla coils. Basically same machine but if it on top of a wire it will also produce energy.
> 
> See [this thread](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=25244) for reference

I'd like to do a gradual difficulty increase, perhaps in future, but not here.

LemonInTheDark commented:
> 
> 
> Here's IMHO better sprites + the green one.
> 
> [beam.zip](https://github.com/tgstation/tgstation/files/4100890/beam.zip)

Yes
[Blessed](https://cdn.discordapp.com/attachments/458452245256601615/669733029375377408/unknown.png)
The percentage is a bug with the monitor, I'll fix it soon.

LemonInTheDark commented:
> Mole can be portal. And power + mole can be singulo. This way normal sabotages still gonna be achievable. And something more memy like the portal thingy can require babysitting.

Gonna test some more, but it seems like the high power setups tend to turn themselves off. This means we can safely set the greens shit behind power + mol, because it will be reasonably harder to achieve. you would have to stabalize between 5000 and 7000 power, or have grounding rods and fend off pyro anomalies.

JAremko commented:
> > I think it will help with testing if this PR included extremely basic merge of grounding rods and tesla coils. Basically same machine but if it on top of a wire it will also produce energy.
> > See [this thread](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=25244) for reference
> 
> I'd like to do a gradual difficulty increase, perhaps in future, but not here.

Is it even possible then to collect power from blue or red arcs with coils? I mean if you surround everything with the rods (and you have to) they will steal all the zaps from coils? Or would they, at least, have equal priority? But even then it won't be intuitive. I think the only way to make it work in a simple and predictable way is to merge the machines

LemonInTheDark commented:
> Is it even possible then to collect power from blue or red arcs with coils? I mean if you surround everything with the rods (and you have to) they will steal all the zaps from coils? Or would they, at least, have equal priority? But even then it won't be intuitive. I think the only way to make it work in a simple and predictable way is to merge the machines

Yes it is, the arc checks for any coils, with a chance at ignoring each one, before checking for rods. So coils have greater priority

JAremko commented:
[portals.zip](https://github.com/tgstation/tgstation/files/4101202/portals.zip)

`¯\_(ツ)_/¯`

JAremko commented:
> Yes it is, the arc checks for any coils, with a chance at ignoring each one, before checking for rods. So coils have greater priority

This is great. Non intuitive as hell but at least works. It's a start :stuck_out_tongue:

JAremko commented:
Wait a second. So the rods are needed simply because zaps will miss them sometimes?  Why not make them never miss and remove rods for the sanity sake  :thinking:

LemonInTheDark commented:
> 
> 
> Wait a second. So the rods are needed simply because zaps will miss them sometimes? Why not make them never miss and remove rods for the sanity sake 🤔

The missing is an intended feature of the sm, keeps the random looking arc paths. It just gives priority to the coils

LemonInTheDark commented:
> 
> 
> [portals.zip](https://github.com/tgstation/tgstation/files/4101202/portals.zip)
> 
> `¯\_(ツ)_/¯`

It'll do.

JAremko commented:
> > Wait a second. So the rods are needed simply because zaps will miss them sometimes? Why not make them never miss and remove rods for the sanity sake thinking
> 
> The missing is an intended feature of the sm, keeps the random looking arc paths. It just gives priority to the coils

To clarify. I meant "So the rods exist simply because zaps sometimes miss coils? Why not make zaps never miss coils and remove rods for the sanity sake ." 

> The missing is an intended feature of the sm, keeps the random looking arc paths.

Does it work the same way for tesla? :thinking: 

- - -- - - -

I think my problem  with all of this is that:
1. Rod/coil interaction is arcane. It wold be logical to assume that rod (given its 100% efficiency) will
steal all the zaps from the covered area but it's not, it has special case for coils.   
2. Coils and rods are basically the same machine except one is generating energy  and catching zaps and the other one catching all the zaps, kinda... (read the first point)
3. rods double the chores and advanced SM build already is like 20+ minutes of preparation.

:thinking:

LemonInTheDark commented:
> To clarify. I meant "So the rods exist simply because zaps sometimes miss coils? Why not make zaps never miss coils and remove rods for the sanity sake ."

I don't know my man, some guy 8 years ago added teslas and here we are. Also rods ground all the charge, whereas coils just eat some of it.

> Does it work the same way for tesla? 🤔

The tesla is deterministic, so it will always target the same things when it's in the same position. 

>     1. Rod/coil interaction is arcane. It wold be logical to assume that rod (given its 100% efficiency) will
>        steal all the zaps from the covered area but it's not, it has special case for coils.

.... :) It also steals from bikes, but that's another story.

>     2. Coils and rods are basically the same machine except one is generating energy  and catching zaps and the other one catching all the zaps, kinda... (read the first point)

Coils catch zaps and then send them on, rods will ground em. 

>     3. rods double the chores(on top of coils) and advanced SM setup already can be like 20+ minutes of preparation.

I can understand this, as I understand it most of the trouble here is constructing/ordering the parts.
I'm ok with it being a longer time investment, it's worth noting you only really need 1 grounding rod as we sit now.

By the way, I'm sorry if I've come off as rude, I really appreciate all the help you've given me.

LemonInTheDark commented:
pushed a rough draft of creatures to add to the portals, thoughts, opinions, passive aggressive pokes?

JAremko commented:
cluwns or felinids with space lube. Hard to choose, maybe both :thinking:

JAremko commented:
Maybe the portals shouldn't be the same each time. Sometimes it's cluwns and other honky stuff. Sometimes something else.

JAremko commented:
Just imagine those two scenarios:

SM goes critical and emits soul shattering HONK that turns everyone in the engineering into cluwns (#irony #instant_carma). SM itself becomes huge clown statue(core) that emits bananium blob that slips everyone, shoots back with bananas. Ah yes! Also ~blobernauts~ cluwne abominations and clown planet portals.

Or

SM turns into suspiciously looking marker, everything dead on the station rises and gets really pissed. Marker spreads zambie blob and portals spawn more zambies.

trollbreeder commented:
> pushed a rough draft of creatures to add to the portals, thoughts, opinions, passive aggressive pokes?

Angry lil' syndicates. They're already a portal storm, so it's easy.

AtlasRules commented:
i never thought my joke about green zaps would come this far

i am proud

TheMidnightRose commented:
I feel like this is in response to my trillion kilowatt overcharge builds. 

But please continue cleaning up Supermatter code. I really want my Admin VV abuse to actually let me creates a Tiberium Crystal event.

JAremko commented:
Blob made out of SM crystal matter that grows from a delaminated SM and must reach the critical mass for the tiberium event. This is actually great reference to C&C, fits the SM fluff nicely and gonna be pretty fun :thinking:

LemonInTheDark commented:
> 
> 
> i never thought my joke about green zaps would come this far
> 
> i am proud

I see good feature request, I bite

LemonInTheDark commented:
> 
> I feel like this is in response to my trillion kilowatt overcharge builds.
> 
> But please continue cleaning up Supermatter code. I really want my Admin VV abuse to actually let me creates a Tiberium Crystal event.

Mostly separated, been working on it for a week or so.
I have seen your sm's tho, like em. Wish you would push the power limit more tho.
Thanks for bringing the good word of tesla coils and grounding rods to the masses.

On another note, I've seen some of your vv sm's, like em. What would you need unshited to make your admeme dreams a reality?

TheMidnightRose commented:
> > I feel like this is in response to my trillion kilowatt overcharge builds.
> > But please continue cleaning up Supermatter code. I really want my Admin VV abuse to actually let me creates a Tiberium Crystal event.
> 
> Mostly separated, been working on it for a week or so.
> I have seen your sm's tho, like em. Wish you would push the power limit more tho.
> Thanks for bringing the good word of tesla coils and grounding rods to the masses.
> 
> On another note, I've seen some of your vv sm's, like em. What would you need unshited to make your admeme dreams a reality?

![image](https://user-images.githubusercontent.com/32805898/73040851-c993be80-3e20-11ea-9a6c-4e3cd7a145db.png)
Power transmission bonus, dynamic powerloss scaling,  powerloss inhibitor and config hallucination power. 
Also heres a screenshot of one of the overcharge builds. https://i.imgur.com/wTXiVla.png

JAremko commented:
It gives me ideas. What if moving SM shard less than 2 tiles away from the big boy SM linked them. Basically they become one like modular machines and all the properties of an original SM got buffed, including bad ones to balance this. Basically it should become more powerful, have more health but also collapse faster  and be in general less stable.  Linking can be visualized as a some kind of an arc between them :thinking: 

###### It's more blackjack than honkers  :wink:

LemonInTheDark commented:
> Power transmission bonus, dynamic powerloss scaling, powerloss inhibitor and config hallucination power.

What's the issue you have with them precisely?

TheMidnightRose commented:
> > Power transmission bonus, dynamic powerloss scaling, powerloss inhibitor and config hallucination power.
> 
> What's the issue you have with them precisely?

Either they do nothing when they are var edited, or they revert automatically back to whatever default value they start with.

JAremko commented:
This is what portals should spawn:
![spawns](https://user-images.githubusercontent.com/1898905/73110782-55384880-3f10-11ea-821b-eea44dcddcad.png)

JAremko commented:
Also the portals should ooze space lube and throw bananas at the crew members :thinking:

JAremko commented:
Or maybe not even lube. The permanently slippery and noky bananium floor should creep out of them like benos weeds thingy. Basically the station gets consumed by the  :clown_face: :earth_africa:

LemonInTheDark commented:
> 
> 
> Or maybe not even lube. The permanently slippery and honky bananium floor should creep out of them like benos weeds thingy.
> 
> Basically the station gets consumed by the 🤡 🌍
> 
> Also mobs and portals might not be aggressive towards clowns so a clown can finally be the hero that station needs and deserves.

Christ we've jumped out of scope and right off a cliff.
I plan on doing a light addition. and then expand on it later based on feedback. 
Need some time to get gud at coding before some of this. 
Will probably make clwne a low chance replacement for general spooky alien shit.

JAremko commented:
I watched some youtube videos and pretended to be pixel artist 

[portals.zip](https://github.com/tgstation/tgstation/files/4111641/portals.zip)

That's kinda draft (I only made the first frame the rest are generated with rotation, for now). Is it better?  :thinking:

LemonInTheDark commented:
> 
> 
> I watched some youtube videos and pretended to be a pixel artist.
> 
> ~[portals.zip](https://github.com/tgstation/tgstation/files/4111641/portals.zip)~ woops frame dupe, this is better:
> ~[portals.zip](https://github.com/tgstation/tgstation/files/4111651/portals.zip)~ woops uploaded the same fing file
> So like probably this one:
> [portals.zip](https://github.com/tgstation/tgstation/files/4111655/portals.zip)
> 
> That's kinda draft (I only made the first frame the rest are generated with rotation, for now). Is it better? 🤔

I like that a good deal more, more vibrant. If you want to change it just post the new one and I will most likely use it.

> 
> Kinda went for R&M parody(given that I'm a parody of a pixel memer). Well the idea is that since we gonna spawn weird shit out of it it should be some weird shit with weird references. The question is will anyone even understand... what is the reference 😄

Listen we're adding a hl1 reference/feature, the whole game is a reference. Hell, the power cap for damage is 9000.

JAremko commented:
Wish I could paint a fucking headcrab :thinking:

JAremko commented:
I guess spawning blob zambies and spores would do.

JAremko commented:
@LaharlMontogmmery  in the case if you figure out how to make it spawn xeno weed:


[xenian_weed.zip](https://github.com/tgstation/tgstation/files/4111690/xenian_weed.zip) Its just benos weed color corrected to match Xen ground texture, [they are quite similar](https://www.youtube.com/watch?v=1_HPtndiiJE&feature=youtu.be&t=331)

This turf, blob spores and blob zambies, some glowshrooms, miasma production and it's gonna be pretty Half-Lifish to me.

JAremko commented:
So like while we are falling off that cliff into the land of pure awesomeness:
1.  Consider using number three in naming as often as possible and grouping things in three.
2. How about making resonance cascade extremely powerful but on top of mole + power delem adding extra gate. One of:
  - It might require busting with a SM shard (as I proposed above)
  - Or it will need a crystal sample item. Basically same as with HL. The item can be uplink device for engineers only or Lavaland ruins spawn.

JAremko commented:
Another sets of spawns can bee benos + beno weed creeping out.  Spawning face hugers or even larvae is totally ok since the shuttle will be called anyway and benos won't have time to reproduce.

JAremko commented:
So like it's THREE types of spawns:
- zambie shit
- benos
- cluwne

JAremko commented:
Hey on top of portals some  bluespace anomalies could be real fitting.

JAremko commented:
Btw AI can say "Gordon Freeman Report To Supermatter We Have An Experiment To Run"  all the TTS are present. I did it couple of times to scare the crew.

JAremko commented:
[portal.zip](https://github.com/tgstation/tgstation/files/4117507/portal.zip)

Welp.  Hopefully it'll do. I think my carrier as an pixel artist is over.  :roll_eyes:

LemonInTheDark commented:
> 
> 
> ~[portal.zip](https://github.com/tgstation/tgstation/files/4117507/portal.zip)~
> [portal.zip](https://github.com/tgstation/tgstation/files/4117567/portal.zip)
> 
> Welp. Hopefully it'll do. I think my carrier as an pixel artist is over. 🙄
> ###### For any kind of a meaningful improvements it has to be started over and each frame has to be hand painted 🤔

Thanks man, that looks great. Frankly you hard carried the art here, thank you so much. Gotta get my head down and code soon.

JAremko commented:
It looks...   You are welcome anyway :smile:

JAremko commented:
@LemonInTheDark  
![crabby](https://user-images.githubusercontent.com/1898905/73220205-9164e680-4166-11ea-8a0b-a09236793e8c.png)

Does it look like a headcrab :thinking:

JAremko commented:
I think all mobs from the portals should get bonus damage from crowbars :thinking:

JAremko commented:
[crabby.zip](https://github.com/tgstation/tgstation/files/4120450/crabby.zip)

I'm a proud father of the Crabby

LemonInTheDark commented:
> 
> 
> [crabby.zip](https://github.com/tgstation/tgstation/files/4120450/crabby.zip)
> 
> I'm a proud father of the Crabby. If it's good enough I'll try making the "hat" sprite, dead sprite and flying sprite. The idea is that it will behave like a spiderling - hide under furniture after spawning but when a potential prey enter the agro range it will launch itself at the prey in the straight line (become projectile) and if missed will enter short cooldown and try again. But if it hits you it will become your hat that constantly deals damage and blinds you, to get it off you have to robust it (preferably with a crowbar) by hitting yourself on the head. And if you die with that thing on you - you become a zambie. Also they prolly should attach themselves to dead bodies 🤔

Yaknow what fuck it lets do this shit. I like the moving sprite well enough, I feel the inactive sprite needs to be different from the first frame of the moving sprite though. maybe some amount of scarring or something? Not sure what would work for a headcrab but it needs something separate.

I'm down for that behavior, I am considering "Atomizing" this pr so the fixes that need to go through to zaps can happened, and then we can half life this bitch.

TheVekter commented:
They're waiting for you, Gordon.

_In the test chamber._

JAremko commented:
[crabby.zip](https://github.com/tgstation/tgstation/files/4121987/crabby.zip)

Fixed the volume change. 
 

> I feel the inactive sprite needs to be different from the first frame of the moving sprite though.

Hmm. I think I have an idea.

BadSS13Player commented:
How did a PR about SM tesla bolts get into brainstorming a resonance/honk cascade event? You should probably open a new PR once you're done brainstorming and coding the event, instead of committing it into this branch/PR. Might get merge conflicts with supermatter.dm when this one gets merged tho, so not sure if that's a good idea.

LemonInTheDark commented:
> 
> 
> How did a PR about SM tesla bolts get into brainstorming a resonance/honk cascade event? You should probably open a new PR once you're done brainstorming and coding the event, instead of committing it into this branch/PR. Might get merge conflicts with supermatter.dm when this one gets merged tho, so not sure if that's a good idea.

This is what I like to call in the business feature creep. I'm thinking on atomizing as I mentioned before, but the hooking the green zaps do relies on my zap cleanup, so I am unsure.

LemonInTheDark commented:
> 
> 
> [crabby.zip](https://github.com/tgstation/tgstation/files/4121987/crabby.zip)
> 
> Fixed the volume change.
> 

You can increase the time each frame spends playing, it seems like what you're trying to do with the active animation.

> > I feel the inactive sprite needs to be different from the first frame of the moving sprite though.
> 
> Hmm. I think I have an idea.

:)

JAremko commented:
@LaharlMontogmmery  how about this idle pose 
[crabby.zip](https://github.com/tgstation/tgstation/files/4126684/crabby.zip)

The ideas is that crabby trying to get as flat as possible to squeeze under a table + his "teeth" are hidden.

LemonInTheDark commented:
> 
> 
> @LaharlMontogmmery how about this idle pose: [crabby.zip](https://github.com/tgstation/tgstation/files/4126708/crabby.zip)
> 
> The idea is that Crabby trying to get as flat as possible to squeeze under a table + his "teeth" are hidden as he's a super chill dude.

I like it better, the variation is much appreciated, it'd be nice if the claws were a bit curved? so it's not perfectly flat, just very close. Could just be my nonspriter midnight brain talking tho.

JAremko commented:
> it'd be nice if the claws were a bit curved?

I'll do it it I find a way to antialias it.

> Could just be my nonspriter midnight brain talking tho.

Same here. no idea what I'm doing :thinking:

JAremko commented:
@LaharlMontogmmery  hello there :thinking:

LemonInTheDark commented:
> Same here. no idea what I'm doing 🤔

M O O D

JAremko commented:
[crabby.zip](https://github.com/tgstation/tgstation/files/4127594/crabby.zip)

Hopefully it is better.

LemonInTheDark commented:
> 
> 
> [crabby.zip](https://github.com/tgstation/tgstation/files/4127713/crabby.zip)
> 
> Hopefully it is better.

Lookin good, just need to do/learn mob code and we should be good.

LemonInTheDark commented:
> 
> Lookin good, just need to do/learn mob code and we should be good.

Oh I'm going to need to stir up a crab projectile icon, could you help with that?

JAremko commented:
sure also dead crab sprite and hat one for zambies

JAremko commented:
@LemonInTheDark  Her's something for you to work with
[crabby.zip](https://github.com/tgstation/tgstation/files/4133416/crabby.zip)

I'll try to clan up flying sprite but should be ok for now.

JAremko commented:
[crabby.zip](https://github.com/tgstation/tgstation/files/4133508/crabby.zip)

:thinking:

LemonInTheDark commented:
> 
> 
> ~[crabby.zip](https://github.com/tgstation/tgstation/files/4133508/crabby.zip)~
> ~[crabby.zip](https://github.com/tgstation/tgstation/files/4133524/crabby.zip)~
> ~[crabby.zip](https://github.com/tgstation/tgstation/files/4133653/crabby.zip)~
> [crabby.zip](https://github.com/tgstation/tgstation/files/4133777/crabby.zip)
> 
> 🤔

I like the flying icon, you think you can do more angles?

JAremko commented:
> I like the flying icon, you think you can do more angles?

Maybe. But then perceptually it might be better to have only this projection, when it turned  towards the viewer like it flies into your face.

JAremko commented:
That's the zambie head (hair)
[crabbed.zip](https://github.com/tgstation/tgstation/files/4139113/crabbed.zip)

The idea is that it will replace character hair when it get's jumped by a crabby. Crabby will start to nom dudes head and deal damage to it but the victim can fight back by bashing itself on the head with, preferably, crowbar. If the poor dude dies its sprite will be zombified (like rameroll zombie) but with the crabby had (hair) :thinking: 

- - - - - -

Left and right sprites are different. I kept them both because I'm not sure which one is better.

JAremko commented:
When a portal spawns something it should create a flash similar to the one's that gold slime cores produce - it's a nice way to "hide" the process. The flash can be green instead of white.

- - - - - - - - -

Cluwne spawns should infect you with a cluwning virus on hit - it will be their alternative to crabbies.

JAremko commented:
[crabbed.zip](https://github.com/tgstation/tgstation/files/4140260/crabbed.zip)

I think this is the best variant.

JAremko commented:
Shaded Crabby a bit more + made flying sprite a lot lot more terrifying
[crabby.zip](https://github.com/tgstation/tgstation/files/4140532/crabby.zip)

LemonInTheDark commented:
> Shaded Crabby a bit more + made flying sprite a lot lot more terrifying
> [crabby.zip](https://github.com/tgstation/tgstation/files/4140532/crabby.zip)

Sweet.

Looking close to complete art wise, I think we need a leaping tell, so players have an idea of when it's jumping at them.

LemonInTheDark commented:
> Shaded Crabby a bit more + made flying sprite a lot lot more terrifying
> [crabby.zip](https://github.com/tgstation/tgstation/files/4140532/crabby.zip)

I like it, and the idea of overlapping hair seems easy enough, I think we need side facing views for it too, could you do that?

JAremko commented:
@LemonInTheDark  it has 4 directions

JAremko commented:
@LemonInTheDark 
[crabby.zip](https://github.com/tgstation/tgstation/files/4142178/crabby.zip)

Added leaping animation. It's basically slightly modified frame of walking "exited" one.

LemonInTheDark commented:
> 
> 
> @LemonInTheDark
> ~[crabby.zip](https://github.com/tgstation/tgstation/files/4142178/crabby.zip)~
> ~[crabby.zip](https://github.com/tgstation/tgstation/files/4142180/crabby.zip)~
> ~[crabby.zip](https://github.com/tgstation/tgstation/files/4142197/crabby.zip)~
> [crabby.zip](https://github.com/tgstation/tgstation/files/4142198/crabby.zip)
> 
> Added leaping animation. It's basically slightly modified frames of walking "exited" one. Should do for now. I might also add more directions for flying one and that's prolly it 🤔

Aight, thank you so much for your help, I'd be sunk without cha. I'm assuming you and OFQ are the same guy. would you like to be credited differently?

JAremko commented:
@LemonInTheDark  No problem :+1:

JAremko commented:
Nah it seems like flying sprite should have only one direction. Other projections are much less effective. Basically it's hard to impress anyone with Crabbies butt  :thinking:

JAremko commented:
@LemonInTheDark I decided to go and embrace the butt, it might be not terrifying but at least it has something to look at.

Do you think it'll work (the back facing flying sprite)
[crabby.zip](https://github.com/tgstation/tgstation/files/4143673/crabby.zip)

JAremko commented:
[crabby.zip](https://github.com/tgstation/tgstation/files/4144512/crabby.zip)

Ok 4 directions for the flying pose.

JAremko commented:
[crabby_contrast.zip](https://github.com/tgstation/tgstation/files/4144792/crabby_contrast.zip)

Slightly more contrast.  Not sure which one is better :thinking:

LemonInTheDark commented:
> 
> 
> [crabby_contrast.zip](https://github.com/tgstation/tgstation/files/4144792/crabby_contrast.zip)
> 
> Slightly more contrast. Not sure which one is better 🤔

More contrast seems good, at least to my coder brain. Having these things pop out more seems like a good decision.

I fucking love the flying sprites, so goddamn much. FUCK I NEED TO FINISH THIS BEFORE IT GETS BETTER.

JAremko commented:
@LemonInTheDark  
> I fucking love the flying sprites, so goddamn much. 

I think it has a problem communicating direction. Maybe this way it will be better [crabby_contrast.zip](https://github.com/tgstation/tgstation/files/4146302/crabby_contrast.zip)

JAremko commented:
While we at it I think rad generation of BZ SM needs a heavy buff. Running something like 60%BZ and the restis n2o + SM farts has a real pathetic energy generation while being a pain in the ass to cool :thinking:

LemonInTheDark commented:
> 
> 
> While we at it I think rad generation of BZ SM needs a heavy buff. Running something like 60% BZ and the rest n2o + SM farts has pathetic energy generation while being a pain in the ass to cool 🤔

B R A H.

Frankly most of the rad output with bz comes from the rad balls. May need some touching, needs some special feature to make it different from trit.

Rather then a zombie sprite, a.... gasmask? sprite would be more fitting. Think facehuggers or whatever looks appropriate to you.

LemonInTheDark commented:
Need to unfuck some of my shitcode and add some icons/oxy damage, but we're well on our way.

JAremko commented:
@LemonInTheDark  You mean replacing "crabbed" sprites with a gas mask?  If there are problems with hair how about replacing entire head when the crab attached (like with the spore zambie) ? I would prefer to avoid adding more overlap between face huger and face crab :stuck_out_tongue:

JAremko commented:
[crabbed.zip](https://github.com/tgstation/tgstation/files/4169599/crabbed.zip)

LemonInTheDark commented:
> 
> 
> Adjusted with an example: [crabbed.zip](https://github.com/tgstation/tgstation/files/4169599/crabbed.zip)

.... Nightcode, forgive me I forgot/missed the head sprites. I thought you had made some but couldn't find any. Having it be more like a hat would work fine, I will strive towards crowbar focused design in all things :).

JAremko commented:
> I will strive towards crowbar focused design in all things :)

Don't forget to overuse number 3 :wink:

JAremko commented:
[crabbed.zip](https://github.com/tgstation/tgstation/files/4169965/crabbed.zip)

So something like this then

JAremko commented:
@LemonInTheDark  color corrected and dumped all into a singe file
[crabby_all.zip](https://github.com/tgstation/tgstation/files/4170113/crabby_all.zip)

JAremko commented:
@LemonInTheDark ok this is it.  I fixed crab hat thingy back view - back legs looked like eyes.  Also I made an extra variant with super eye bleeding colors (so the archive has 2 files) [crabs.zip](https://github.com/tgstation/tgstation/files/4170280/crabs.zip)

JAremko commented:
Fixed the problem with flying crabby orientation and how it embeds onto host's head [crabs.zip](https://github.com/tgstation/tgstation/files/4171763/crabs.zip)

LemonInTheDark commented:
> 
> 
> Fixed the problem with flying crabby orientation and how it embeds onto host's head [crabs.zip](https://github.com/tgstation/tgstation/files/4171763/crabs.zip)

Fuck those side sprites look good.

LemonInTheDark commented:
Opening a pr for crabby soon, just need to write it up now

LemonInTheDark commented:
Aight, ready for review. I'm keeping the green bolts and alert message because they seem like fine enough easter eggs to me even separate from portals and shit.

JAremko commented:
I think this was the last portal sprite revision, had it dig deep into the comment heap 
:smile:  https://github.com/tgstation/tgstation/files/4117567/portal.zip

LemonInTheDark commented:
Aight yeeted the green bolts and alert message.

optimumtact commented on code/modules/power/supermatter/supermatter.dm:
```diff
@@ -2,6 +2,16 @@
+#define LIVING (ROD - 1)
+#define MACHINERY (LIVING - 1)
+#define OBJECT (MACHINERY - 1)
+#define LOWEST (OBJECT - 1)
```
should this not go the otherway? otherwise this will start being negative at some point cause of someone not counting properly

LemonInTheDark replied:
being negative doesn't actually effect the code, although going bottom up does seem a little more clean, so sure.

Ghilker commented:
can't wait for this to be ready *_* like all your other stuff Lemon!

LemonInTheDark commented:
It's ready as of now, just flooded with comments. Glad to keep ya waiting Ghilker.

TheMidnightRose commented:
I cannot wait to horrifically destroy the station with this.

TheMidnightRose commented:
So now that the Tesla Coils cannot infinitely chain arcs and has been solved in another PR. Is this still fully ready to be merged soon?

LemonInTheDark commented:
Need to solve the minor merge conflicts, but aside from that, yes it is still ready.

LemonInTheDark commented:
I've pulled out all the non essential changes from this pr, an atomized version will be up soon. I've kept a few things between both versions because of how important they are. Think common runtimes, ect.

LemonInTheDark commented:
not again

LemonInTheDark commented:
kill me.

LemonInTheDark committed:
Adds blue and red zaps, which require 7000 and 9000 eer each. The blue zaps hurt and emp objects, along with uncapping zap damage. The red zaps explode machines and stun carbons

LemonInTheDark commented:
Ahhhh a clean history....

JAremko commented:
At some point the commit log had more lines that the whole PR :smile:

LemonInTheDark commented:
Too much shit in 1 pr. Life is a wild ride

> 
> 
> At some point the commit log had more lines that the whole PR 😄

My job is never done. Knowing me the testmerge will show some insanity that only players can produce.

kriskog commented on code/modules/power/supermatter/supermatter.dm:
```diff
@@ -544,8 +546,16 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
+					//It'll stun more now, and damage will hit harder, gloves are no garentee.
+					//Machines go boom
+					flags |= (ZAP_MOB_STUN | ZAP_MACHINE_EXPLOSIVE)
 					zap_count += 1
```
This should probably be a switch or you are assigning vars 3 times every zap.

kriskog commented on code/modules/power/supermatter/supermatter.dm:
```diff
@@ -972,12 +982,20 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
+			if(power > SEVERE_POWER_PENALTY_THRESHOLD)
+				multi *= 2
+				if(power > CRITICAL_POWER_PENALTY_THRESHOLD)
+					multi *= 2
```
This one too

LemonInTheDark commented:
I've replaced the if statements with switches. I thought this would be less clear but you're right, it seems more obvious.

I've updated the damage comment. The zap damage is capped by the input power clamp. This might be worth looking into later, but not now.

The things for me to keep an eye out for during testmerge are the emps and the red zaps, they may need to explode more then 20% of the time.

I'd like to avoid making weaponizing high power delams impossible with the explosions and emps, maybe just a bit more complex, with some new upsides.
